### PR TITLE
Stack: Modifying examples so they don't use mergeStyles, avoiding collision with slots/tokens work

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

Stack: Modifying examples so they don't use mergeStyles, avoiding collision with slots/tokens work.

### CODE REVIEW

File: packages/experiments/src/components/Stack/examples/Stack.Horizontal.Basic.Example.tsx [hunk 1]
```diff
@@ -1,62 +1,67 @@
...
 
-      item: {
+    const itemStyles: IComponentStyles<IStackItemSlots> = {
+      root: {
         color: DefaultPalette.white,
         background: DefaultPalette.themePrimary,
         padding: 5
       }
-    });
+    };
```
Comment (dzearing) [inline packages/experiments/src/components/Stack/examples/Stack.Horizontal.Basic.Example.tsx]: Can these const values be moved outside of the class?

File: packages/experiments/src/components/Stack/examples/Stack.Vertical.Grow.Example.tsx [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
         color: DefaultPalette.white
       }
-    });
+    };
 
     return (
       <Stack gap={5}>
-        <Stack gap={5} padding={10} className={styles.root}>
-          <Stack.Item grow={3} className={styles.item}>
+        <Stack gap={5} padding={10} styles={rootStyles}>
```
Comment (dzearing) [inline packages/experiments/src/components/Stack/examples/Stack.Vertical.Grow.Example.tsx]: I'd really recommend we avoid suggesting using styles like this for Stack examples. It deemphasizes the important parts of stack.

File: packages/experiments/src/components/Stack/examples/Stack.Vertical.Spacing.Example.tsx [hunk 3]
```diff
@@ -19,73 +24,73 @@ export class VerticalStackSpacingExample extends React.Component<{}, {}> {
...
       <Stack gap={10}>
         <Stack horizontal gap={50}>
           <Stack>
             <span>Numerical spacing</span>
-            <Stack className={styles.root} gap={10} padding={10}>
-              <span className={styles.item}>1</span>
-              <span className={styles.item}>2</span>
-              <span className={styles.item}>3</span>
+            <Stack styles={rootStyles} gap={10} padding={10}>
+              <span className={mergeStyleSets(itemStyles).root}>1</span>
```
Comment (dzearing) [inline packages/experiments/src/components/Stack/examples/Stack.Vertical.Spacing.Example.tsx]: this feels really gross...

You do know about `mergeStyles` right? You don't need to use `mergeStyleSets` super redundantly like this!

File: packages/experiments/src/components/Stack/examples/Stack.Vertical.WrapNested.Example.tsx [hunk 4]
```diff
@@ -88,13 +96,13 @@ export class VerticalStackWrapNestedExample extends React.Component<{}, IExample
             <span>7</span>
           </Stack>
 
-          <Stack wrap gap={50} verticalGap={20} className={styles.stackTwo}>
+          <Stack wrap gap={50} verticalGap={20} styles={stackTwoStyles}>
```
Comment (dzearing) [inline packages/experiments/src/components/Stack/examples/Stack.Vertical.WrapNested.Example.tsx]: can we just get rid of all these inline styles?

**Comment by JasonGore** [comment]
Just to be super clear here, these examples should continue to work as they were regardless of slots/tokens work. I consider the fact that they don't work bugs with slots/tokens that I need to resolve. I only called out styles as an alternative (and more consistent and conventional) approach with our other examples in general.

**Comment by khmakoto** [comment]
@dzearing like @JasonGore mentioned, I'm making this change because the styles were being busted after the slots/token work (based on working off Jason's branch). Maybe I should close this PR and leave it as it was before and just help @JasonGore figure out why is Stack style being busted by token's work.

**Pull request closed without merging**